velox/vector/DictionaryVector.h

```diff
@@ -152,7 +152,20 @@ class DictionaryVector : public SimpleVector<T> {
   }
 
   BaseVector* loadedVector() override {
-    dictionaryValues_ = BaseVector::loadedVectorShared(dictionaryValues_);
+    if (initialized_) {
+      return this;
+    }
+
+    SelectivityVector rows(dictionaryValues_->size(), false);
+    for (vector_size_t i = 0; i < this->size(); i++) {
+      if (!BaseVector::isNullAt(i)) {
```

Comment:
I believe using initialized_ flag already accomplished that. We can optimize this loop for the case when there are no nulls, but it is not clear whether it would be beneficial.

Reply:
Thanks Masha , I currently believe initialize_ flag is set only if allloaded_ is true; I am not sure if ensureLoadedRows sets that to true; Patrick please correct me if thats the case.

Reply:
@kgpai Yes, ensureLoadedRows sets allLoaded_ flag in the LazyVector to `true`.

```
  void load(RowSet rows, ValueHook* hook) const {
    VELOX_CHECK(!allLoaded_, "A LazyVector can be loaded at most once");
    allLoaded_ = true;
```

Reply:
@kgpai is your concern that there might be a case where DictionaryVector::initialized_ is true but LazyVector::allLoaded is not set? In the case of a nested DictionaryVectors by loading inner Dictionary allLoaded should be set (ensureLoadedRows calls LazyVector::load which sets it) and then sets DictionaryVector::initialized_ to true. Please let me know if this clarifies it or if there is another concern.

Reply:
Thanks Patrick, that clarifies it. If initialized_ is always true when allloaded is true and I assume vice versa, is there a reason why we need both ? Can we get rid of one of them to simplify things ?

Reply:
Good question, I guess those are states in different Vector types that are also used independently, but I don't have enough deep understanding of all the use cases. @mbasmanova do you think it makes sense to deduplicate those variables (Dictionary::initialized_, LazyVector::allLoaded)? If so my suggestion would be to do it in a separate PR.